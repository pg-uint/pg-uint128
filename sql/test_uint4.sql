-- Testing uint4

-- From string block

-- NULL
SELECT NULL::text::uint4;
-- Empty
SELECT ''::text::uint4;
-- Zero
SELECT '0'::text::uint4;
-- Min
SELECT '0'::text::uint4;
-- Max
SELECT '4294967295'::text::uint4;
-- Underflow
SELECT '-1'::uint4;
-- Overflow
SELECT '4294967296'::uint4;

-- Ops block

SELECT 1::uint4 = 1::uint4;
SELECT 1::uint4 = 0::uint4;
SELECT 1::uint4 = 1::int2;
SELECT 1::uint4 = 0::int2;
SELECT 1::int2 = 1::uint4;
SELECT 1::int2 = 0::uint4;
SELECT 1::uint4 = 1::int4;
SELECT 1::uint4 = 0::int4;
SELECT 1::int4 = 1::uint4;
SELECT 1::int4 = 0::uint4;
SELECT 1::uint4 = 1::int8;
SELECT 1::uint4 = 0::int8;
SELECT 1::int8 = 1::uint4;
SELECT 1::int8 = 0::uint4;

SELECT 1::uint4 <> 0::uint4;
SELECT 0::uint4 <> 0::uint4;
SELECT 1::uint4 <> 0::int2;
SELECT 0::uint4 <> 0::int2;
SELECT 1::int2 <> 0::uint4;
SELECT 0::int2 <> 0::uint4;
SELECT 1::uint4 <> 0::int4;
SELECT 0::uint4 <> 0::int4;
SELECT 1::int4 <> 0::uint4;
SELECT 0::int4 <> 0::uint4;
SELECT 1::uint4 <> 0::int8;
SELECT 0::uint4 <> 0::int8;
SELECT 1::int8 <> 0::uint4;
SELECT 0::int8 <> 0::uint4;

SELECT 1::uint4 > 0::uint4;
SELECT 0::uint4 > 1::uint4;
SELECT 1::uint4 > 0::int2;
SELECT 0::uint4 > 1::int2;
SELECT 1::int2 > 0::uint4;
SELECT 0::int2 > 1::uint4;
SELECT 1::uint4 > 0::int4;
SELECT 0::uint4 > 1::int4;
SELECT 1::int4 > 0::uint4;
SELECT 0::int4 > 1::uint4;
SELECT 1::uint4 > 0::int8;
SELECT 0::uint4 > 1::int8;
SELECT 1::int8 > 0::uint4;
SELECT 0::int8 > 1::uint4;

SELECT 0::uint4 < 1::uint4;
SELECT 1::uint4 < 0::uint4;
SELECT 0::uint4 < 1::int2;
SELECT 1::uint4 < 0::int2;
SELECT 0::int2 < 1::uint4;
SELECT 1::int2 < 0::uint4;
SELECT 0::uint4 < 1::int4;
SELECT 1::uint4 < 0::int4;
SELECT 0::int4 < 1::uint4;
SELECT 1::int4 < 0::uint4;
SELECT 0::uint4 < 1::int8;
SELECT 1::uint4 < 0::int8;
SELECT 0::int8 < 1::uint4;
SELECT 1::int8 < 0::uint4;

SELECT 1::uint4 >= 0::uint4;
SELECT 0::uint4 >= 1::uint4;
SELECT 1::uint4 >= 1::uint4;
SELECT 1::uint4 >= 0::int2;
SELECT 0::uint4 >= 1::int2;
SELECT 1::uint4 >= 1::int2;
SELECT 1::int2 >= 0::uint4;
SELECT 0::int2 >= 1::uint4;
SELECT 1::int2 >= 1::uint4;
SELECT 1::uint4 >= 0::int4;
SELECT 0::uint4 >= 1::int4;
SELECT 1::uint4 >= 1::int4;
SELECT 1::int4 >= 0::uint4;
SELECT 0::int4 >= 1::uint4;
SELECT 1::int4 >= 1::uint4;
SELECT 1::uint4 >= 0::int8;
SELECT 0::uint4 >= 1::int8;
SELECT 1::uint4 >= 1::int8;
SELECT 1::int8 >= 0::uint4;
SELECT 0::int8 >= 1::uint4;
SELECT 1::int8 >= 1::uint4;

SELECT 0::uint4 <= 1::uint4;
SELECT 1::uint4 <= 0::uint4;
SELECT 1::uint4 <= 1::uint4;
SELECT 0::uint4 <= 1::int2;
SELECT 1::uint4 <= 0::int2;
SELECT 1::uint4 <= 1::int2;
SELECT 0::int2 <= 1::uint4;
SELECT 1::int2 <= 0::uint4;
SELECT 1::int2 <= 1::uint4;
SELECT 0::uint4 <= 1::int4;
SELECT 1::uint4 <= 0::int4;
SELECT 1::uint4 <= 1::int4;
SELECT 0::int4 <= 1::uint4;
SELECT 1::int4 <= 0::uint4;
SELECT 1::int4 <= 1::uint4;
SELECT 0::uint4 <= 1::int8;
SELECT 1::uint4 <= 0::int8;
SELECT 1::uint4 <= 1::int8;
SELECT 0::int8 <= 1::uint4;
SELECT 1::int8 <= 0::uint4;
SELECT 1::int8 <= 1::uint4;

SELECT 120::uint4 + 10::uint4;
SELECT 120::uint4 + 10::int2;
SELECT 4294967295::uint4 + 1::int2;
SELECT 120::int2 + 10::uint4;
SELECT 32767::int2 + 1::uint4;
SELECT (-120)::int2 + 10::uint4;
SELECT 120::uint4 + 10::int4;
SELECT 4294967295::uint4 + 1::int4;
SELECT 120::int4 + 10::uint4;
SELECT 2147483647::int4 + 1::uint4;
SELECT (-120)::int4 + 10::uint4;
SELECT 120::uint4 + 10::int8;
SELECT 4294967295::uint4 + 1::int8;
SELECT 120::int8 + 10::uint4;
SELECT 9223372036854775807::int8 + 1::uint4;
SELECT (-120)::int8 + 10::uint4;

SELECT 120::uint4 - 10::uint4;
SELECT 120::uint4 - 10::int2;
SELECT 0::uint4 - 1::int2;
SELECT 120::int2 - 10::uint4;
SELECT (-32768)::int2 - 1::uint4;
SELECT (-120)::int2 - 10::uint4;
SELECT 120::uint4 - 10::int4;
SELECT 0::uint4 - 1::int4;
SELECT 120::int4 - 10::uint4;
SELECT (-2147483648)::int4 - 1::uint4;
SELECT (-120)::int4 - 10::uint4;
SELECT 120::uint4 - 10::int8;
SELECT 0::uint4 - 1::int8;
SELECT 120::int8 - 10::uint4;
SELECT (-9223372036854775808)::int8 - 1::uint4;
SELECT (-120)::int8 - 10::uint4;

SELECT 120::uint4 * 10::uint4;
SELECT 120::uint4 * 10::int2;
SELECT 4294967295::uint4 * 2::int2;
SELECT 120::int2 * 10::uint4;
SELECT 32767::int2 * 2::uint4;
SELECT (-120)::int2 * 10::uint4;
SELECT 120::uint4 * 10::int4;
SELECT 4294967295::uint4 * 2::int4;
SELECT 120::int4 * 10::uint4;
SELECT 2147483647::int4 * 2::uint4;
SELECT (-120)::int4 * 10::uint4;
SELECT 120::uint4 * 10::int8;
SELECT 4294967295::uint4 * 2::int8;
SELECT 120::int8 * 10::uint4;
SELECT 9223372036854775807::int8 * 2::uint4;
SELECT (-120)::int8 * 10::uint4;

SELECT 120::uint4 / 10::uint4;
SELECT 120::uint4 / 10::int2;
SELECT 1::uint4 / 0::int2;
SELECT 120::int2 / 10::uint4;
SELECT 1::int2 / 0::uint4;
SELECT (-120)::int2 / 10::uint4;
SELECT 120::uint4 / 10::int4;
SELECT 1::uint4 / 0::int4;
SELECT 120::int4 / 10::uint4;
SELECT 1::int4 / 0::uint4;
SELECT (-120)::int4 / 10::uint4;
SELECT 120::uint4 / 10::int8;
SELECT 1::uint4 / 0::int8;
SELECT 120::int8 / 10::uint4;
SELECT 1::int8 / 0::uint4;
SELECT (-120)::int8 / 10::uint4;

SELECT 120::uint4 % 10::uint4;
SELECT 120::uint4 % 10::int2;
SELECT 1::uint4 % 0::int2;
SELECT 120::int2 % 10::uint4;
SELECT 1::int2 % 0::uint4;
SELECT (-120)::int2 % 10::uint4;
SELECT 120::uint4 % 10::int4;
SELECT 1::uint4 % 0::int4;
SELECT 120::int4 % 10::uint4;
SELECT 1::int4 % 0::uint4;
SELECT (-120)::int4 % 10::uint4;
SELECT 120::uint4 % 10::int8;
SELECT 1::uint4 % 0::int8;
SELECT 120::int8 % 10::uint4;
SELECT 1::int8 % 0::uint4;
SELECT (-120)::int8 % 10::uint4;

-- Agg ops block

SELECT sum(s::uint4) FROM generate_series(1, 10000) s;
SELECT avg(s::uint4) FROM generate_series(1, 10000) s;
SELECT min(s::uint4) FROM generate_series(1, 10000) s;
SELECT max(s::uint4) FROM generate_series(1, 10000) s;

-- Generate series block

SELECT s FROM generate_series(1::uint4, 10::uint4) s;
SELECT s FROM generate_series(1::uint4, 10::uint4, 2::uint4) s;

-- Ranges block

SELECT uint4range(0::uint4, 10::uint4);
SELECT uint4range(0::uint4, 4294967295::uint4);
SELECT uint4range(0::uint4, 4294967295::uint4, '[]');
SELECT upper(uint4range(0::uint4, 10::uint4));
SELECT lower(uint4range(0::uint4, 10::uint4));
SELECT isempty(uint4range(0::uint4, 10::uint4));
SELECT uint4range(0::uint4, 10::uint4) @> 9::uint4;
SELECT uint4range(0::uint4, 10::uint4) @> 10::uint4;
SELECT uint4range(0::uint4, 10::uint4) && uint4range(10::uint4,20::uint4);
SELECT uint4range(0::uint4, 10::uint4) && uint4range(9::uint4,20::uint4);
SELECT uint4range(5::uint4, 10::uint4) - uint4range(5::uint4, 10::uint4);
SELECT uint4range(5::uint4, 10::uint4) - uint4range(5::uint4, 9::uint4);
CREATE TEMPORARY TABLE test_uint4range (
    r uint4range,

    EXCLUDE USING GIST (r WITH &&)
);

INSERT INTO test_uint4range (r) VALUES (uint4range(0::uint4, 10::uint4));
INSERT INTO test_uint4range (r) VALUES (uint4range(10::uint4, 20::uint4));
INSERT INTO test_uint4range (r) VALUES (uint4range(19::uint4, 30::uint4));

DROP TABLE test_uint4range;
