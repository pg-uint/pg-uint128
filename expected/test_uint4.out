-- Testing uint4
-- From string block
-- NULL
SELECT NULL::text::uint4;
 uint4 
-------
 
(1 row)

-- Empty
SELECT ''::text::uint4;
ERROR:  invalid input syntax for type uint4: ""
-- Zero
SELECT '0'::text::uint4;
 uint4 
-------
 0
(1 row)

-- Min
SELECT '0'::text::uint4;
 uint4 
-------
 0
(1 row)

-- Max
SELECT '4294967295'::text::uint4;
   uint4    
------------
 4294967295
(1 row)

-- Underflow
SELECT '-1'::uint4;
ERROR:  invalid input syntax for type uint4: "-1"
LINE 1: SELECT '-1'::uint4;
               ^
-- Overflow
SELECT '4294967296'::uint4;
ERROR:  uint4 out of range
LINE 1: SELECT '4294967296'::uint4;
               ^
-- Ops block
SELECT 1::uint4 = 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 = 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 = 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 = 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 = 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 = 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 = 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 = 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 = 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 = 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 = 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 = 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 = 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 = 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <> 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <> 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <> 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <> 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 <> 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 <> 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <> 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <> 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 <> 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 <> 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <> 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <> 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 <> 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 <> 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 > 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 > 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 > 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 > 1::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 > 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 > 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 > 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 > 1::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 > 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 > 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 > 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 > 1::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 > 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 > 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint4 < 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 < 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint4 < 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 < 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 0::int2 < 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 < 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint4 < 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 < 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 0::int4 < 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 < 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint4 < 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 < 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 0::int8 < 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 < 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 >= 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 >= 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 >= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 >= 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 >= 1::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 >= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 >= 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 >= 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 >= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 >= 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 >= 1::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 >= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 >= 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 >= 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 >= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 >= 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 >= 1::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 >= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 >= 0::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 >= 1::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 >= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 <= 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 <= 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 <= 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 <= 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 <= 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint4 <= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint4 <= 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint4 <= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 <= 0::uint4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 <= 1::uint4;
 ?column? 
----------
 t
(1 row)

SELECT 120::uint4 + 10::uint4;
 ?column? 
----------
 130
(1 row)

SELECT 120::uint4 + 10::int2;
 ?column? 
----------
 130
(1 row)

SELECT 4294967295::uint4 + 1::int2;
ERROR:  uint4 out of range
SELECT 120::int2 + 10::uint4;
 ?column? 
----------
      130
(1 row)

SELECT 32767::int2 + 1::uint4;
ERROR:  int2 out of range
SELECT (-120)::int2 + 10::uint4;
 ?column? 
----------
     -110
(1 row)

SELECT 120::uint4 + 10::int4;
 ?column? 
----------
 130
(1 row)

SELECT 4294967295::uint4 + 1::int4;
ERROR:  uint4 out of range
SELECT 120::int4 + 10::uint4;
 ?column? 
----------
      130
(1 row)

SELECT 2147483647::int4 + 1::uint4;
ERROR:  int4 out of range
SELECT (-120)::int4 + 10::uint4;
 ?column? 
----------
     -110
(1 row)

SELECT 120::uint4 + 10::int8;
 ?column? 
----------
 130
(1 row)

SELECT 4294967295::uint4 + 1::int8;
ERROR:  uint4 out of range
SELECT 120::int8 + 10::uint4;
 ?column? 
----------
      130
(1 row)

SELECT 9223372036854775807::int8 + 1::uint4;
ERROR:  int8 out of range
SELECT (-120)::int8 + 10::uint4;
 ?column? 
----------
     -110
(1 row)

SELECT 120::uint4 - 10::uint4;
 ?column? 
----------
 110
(1 row)

SELECT 120::uint4 - 10::int2;
 ?column? 
----------
 110
(1 row)

SELECT 0::uint4 - 1::int2;
ERROR:  uint4 out of range
SELECT 120::int2 - 10::uint4;
 ?column? 
----------
      110
(1 row)

SELECT (-32768)::int2 - 1::uint4;
ERROR:  int2 out of range
SELECT (-120)::int2 - 10::uint4;
 ?column? 
----------
     -130
(1 row)

SELECT 120::uint4 - 10::int4;
 ?column? 
----------
 110
(1 row)

SELECT 0::uint4 - 1::int4;
ERROR:  uint4 out of range
SELECT 120::int4 - 10::uint4;
 ?column? 
----------
      110
(1 row)

SELECT (-2147483648)::int4 - 1::uint4;
ERROR:  int4 out of range
SELECT (-120)::int4 - 10::uint4;
 ?column? 
----------
     -130
(1 row)

SELECT 120::uint4 - 10::int8;
 ?column? 
----------
 110
(1 row)

SELECT 0::uint4 - 1::int8;
ERROR:  uint4 out of range
SELECT 120::int8 - 10::uint4;
 ?column? 
----------
      110
(1 row)

SELECT (-9223372036854775808)::int8 - 1::uint4;
ERROR:  int8 out of range
SELECT (-120)::int8 - 10::uint4;
 ?column? 
----------
     -130
(1 row)

SELECT 120::uint4 * 10::uint4;
 ?column? 
----------
 1200
(1 row)

SELECT 120::uint4 * 10::int2;
 ?column? 
----------
 1200
(1 row)

SELECT 4294967295::uint4 * 2::int2;
ERROR:  uint4 out of range
SELECT 120::int2 * 10::uint4;
 ?column? 
----------
     1200
(1 row)

SELECT 32767::int2 * 2::uint4;
ERROR:  int2 out of range
SELECT (-120)::int2 * 10::uint4;
 ?column? 
----------
    -1200
(1 row)

SELECT 120::uint4 * 10::int4;
 ?column? 
----------
 1200
(1 row)

SELECT 4294967295::uint4 * 2::int4;
ERROR:  uint4 out of range
SELECT 120::int4 * 10::uint4;
 ?column? 
----------
     1200
(1 row)

SELECT 2147483647::int4 * 2::uint4;
ERROR:  int4 out of range
SELECT (-120)::int4 * 10::uint4;
 ?column? 
----------
    -1200
(1 row)

SELECT 120::uint4 * 10::int8;
 ?column? 
----------
 1200
(1 row)

SELECT 4294967295::uint4 * 2::int8;
ERROR:  uint4 out of range
SELECT 120::int8 * 10::uint4;
 ?column? 
----------
     1200
(1 row)

SELECT 9223372036854775807::int8 * 2::uint4;
ERROR:  int8 out of range
SELECT (-120)::int8 * 10::uint4;
 ?column? 
----------
    -1200
(1 row)

SELECT 120::uint4 / 10::uint4;
 ?column? 
----------
 12
(1 row)

SELECT 120::uint4 / 10::int2;
 ?column? 
----------
 12
(1 row)

SELECT 1::uint4 / 0::int2;
ERROR:  division by zero
SELECT 120::int2 / 10::uint4;
 ?column? 
----------
       12
(1 row)

SELECT 1::int2 / 0::uint4;
ERROR:  division by zero
SELECT (-120)::int2 / 10::uint4;
 ?column? 
----------
        0
(1 row)

SELECT 120::uint4 / 10::int4;
 ?column? 
----------
 12
(1 row)

SELECT 1::uint4 / 0::int4;
ERROR:  division by zero
SELECT 120::int4 / 10::uint4;
 ?column? 
----------
       12
(1 row)

SELECT 1::int4 / 0::uint4;
ERROR:  division by zero
SELECT (-120)::int4 / 10::uint4;
 ?column? 
----------
        0
(1 row)

SELECT 120::uint4 / 10::int8;
 ?column? 
----------
 12
(1 row)

SELECT 1::uint4 / 0::int8;
ERROR:  division by zero
SELECT 120::int8 / 10::uint4;
 ?column? 
----------
       12
(1 row)

SELECT 1::int8 / 0::uint4;
ERROR:  division by zero
SELECT (-120)::int8 / 10::uint4;
 ?column? 
----------
        0
(1 row)

SELECT 120::uint4 % 10::uint4;
 ?column? 
----------
 0
(1 row)

SELECT 120::uint4 % 10::int2;
 ?column? 
----------
 0
(1 row)

SELECT 1::uint4 % 0::int2;
ERROR:  division by zero
SELECT 120::int2 % 10::uint4;
 ?column? 
----------
        0
(1 row)

SELECT 1::int2 % 0::uint4;
ERROR:  division by zero
SELECT (-120)::int2 % 10::uint4;
 ?column? 
----------
     -120
(1 row)

SELECT 120::uint4 % 10::int4;
 ?column? 
----------
 0
(1 row)

SELECT 1::uint4 % 0::int4;
ERROR:  division by zero
SELECT 120::int4 % 10::uint4;
 ?column? 
----------
        0
(1 row)

SELECT 1::int4 % 0::uint4;
ERROR:  division by zero
SELECT (-120)::int4 % 10::uint4;
 ?column? 
----------
     -120
(1 row)

SELECT 120::uint4 % 10::int8;
 ?column? 
----------
 0
(1 row)

SELECT 1::uint4 % 0::int8;
ERROR:  division by zero
SELECT 120::int8 % 10::uint4;
 ?column? 
----------
        0
(1 row)

SELECT 1::int8 % 0::uint4;
ERROR:  division by zero
SELECT (-120)::int8 % 10::uint4;
 ?column? 
----------
     -120
(1 row)

-- Agg ops block
SELECT sum(s::uint4) FROM generate_series(1, 10000) s;
   sum    
----------
 50005000
(1 row)

SELECT avg(s::uint4) FROM generate_series(1, 10000) s;
          avg          
-----------------------
 5000.5000000000000000
(1 row)

SELECT min(s::uint4) FROM generate_series(1, 10000) s;
 min 
-----
 1
(1 row)

SELECT max(s::uint4) FROM generate_series(1, 10000) s;
  max  
-------
 10000
(1 row)

-- Generate series block
SELECT s FROM generate_series(1::uint4, 10::uint4) s;
 s  
----
 1
 2
 3
 4
 5
 6
 7
 8
 9
 10
(10 rows)

SELECT s FROM generate_series(1::uint4, 10::uint4, 2::uint4) s;
 s 
---
 1
 3
 5
 7
 9
(5 rows)

-- Ranges block
SELECT uint4range(0::uint4, 10::uint4);
 uint4range 
------------
 [0,10)
(1 row)

SELECT uint4range(0::uint4, 4294967295::uint4);
   uint4range   
----------------
 [0,4294967295)
(1 row)

SELECT uint4range(0::uint4, 4294967295::uint4, '[]');
ERROR:  uint4 out of range
SELECT upper(uint4range(0::uint4, 10::uint4));
 upper 
-------
 10
(1 row)

SELECT lower(uint4range(0::uint4, 10::uint4));
 lower 
-------
 0
(1 row)

SELECT isempty(uint4range(0::uint4, 10::uint4));
 isempty 
---------
 f
(1 row)

SELECT uint4range(0::uint4, 10::uint4) @> 9::uint4;
 ?column? 
----------
 t
(1 row)

SELECT uint4range(0::uint4, 10::uint4) @> 10::uint4;
 ?column? 
----------
 f
(1 row)

SELECT uint4range(0::uint4, 10::uint4) && uint4range(10::uint4,20::uint4);
 ?column? 
----------
 f
(1 row)

SELECT uint4range(0::uint4, 10::uint4) && uint4range(9::uint4,20::uint4);
 ?column? 
----------
 t
(1 row)

SELECT uint4range(5::uint4, 10::uint4) - uint4range(5::uint4, 10::uint4);
 ?column? 
----------
 empty
(1 row)

SELECT uint4range(5::uint4, 10::uint4) - uint4range(5::uint4, 9::uint4);
 ?column? 
----------
 [9,10)
(1 row)

CREATE TEMPORARY TABLE test_uint4range (
    r uint4range,
    EXCLUDE USING GIST (r WITH &&)
);
INSERT INTO test_uint4range (r) VALUES (uint4range(0::uint4, 10::uint4));
INSERT INTO test_uint4range (r) VALUES (uint4range(10::uint4, 20::uint4));
INSERT INTO test_uint4range (r) VALUES (uint4range(19::uint4, 30::uint4));
ERROR:  conflicting key value violates exclusion constraint "test_uint4range_r_excl"
DETAIL:  Key (r)=([19,30)) conflicts with existing key (r)=([10,20)).
DROP TABLE test_uint4range;
